src/parser/type_checking.c

```diff
@@ -20363,6 +20401,20 @@ pt_semantic_type (PARSER_CONTEXT * parser, PT_NODE * tree, SEMANTIC_CHK_INFO * s
       tree = NULL;
     }
 
+  /* When qo_reduce_equality_terms is executed in mq_optimize, a removable predicate like '1=1' is generated.
```

Comment:
This routine looks not releated to this issue. It's better to merge from develop branch.

Reply:
yes. I did